### ISSUE TITLE
(feat req) OpenMP acceleration in parallel_for

### ISSUE THREAD

**Issue by crusaderky** [harrism/hemi#20]
It would be great if, when CUDA is disabled, hemi::paralell_for ran on all available CPUs.
I tried adding a simple #pragma omp for myself, but for some reason OpenMP refused to cooperate...

**Comment by harrism** [comment]
I agree. I would like to look at this in the future if somebody doesn't get to it before me.

**Comment by nitro44x** [comment]
does this meet the requirement?

https://github.com/louietsai/hemi/commit/216f912d87f72f13ca8d7463ce21b275a634a452
